### PR TITLE
fixes #151, add custom solana-py RPC error handling

### LINKED ISSUE #151: Custom solana-py exceptions
Has it been considered to raise exceptions custom to solana-py library instead of generic HTTPError created in:

https://github.com/michaelhly/solana-py/blob/f38f01a3c34ddc2d2f67a56d104246e458644717/src/solana/rpc/providers/core.py#L54

This may also be useful in timeout scenario. It would be a simple change to wrap provider code in decorator catching and converting into custom exception. Then clients and classes composed with solana library can more specifically catch exceptions.

Would be happy to create PR, let me know your thoughts.

### PR DESCRIPTION
**Pull request by broper2**

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -118,6 +118,20 @@ def freeze_authority() -> Keypair:
...
+    return client
+
+
+@pytest.fixture(scope="session")
+def unit_test_http_client_async() -> AsyncClient:
+    """Async client to be used in unit tests."""
+    client = AsyncClient(commitment=Processed)
+    return client
+
+
```
Comment (michaelhly) [inline tests/conftest.py:134]: Is it possible to use 
https://github.com/michaelhly/solana-py/pull/152/files#diff-e52e4ddd58b7ef887ab03c04116e676f6280b824ab7469d5d3080e5cba4f2128R15-R20

instead?

Comment (broper2) [reply tests/conftest.py:134]: I do not see this used anywhere yet, little confused. Are you suggesting I combine these proposed unit test fixtures into one, exposing the named tuple around a sync, async, and event loop?

Comment (michaelhly) [reply tests/conftest.py:134]: @broper2 sorry. I misread your test. Please move it to integration tests. Unit tests should not be making rpc requests.

Comment (broper2) [reply tests/conftest.py:134]: @michaelhly in the tests, I am mocking the request call. It is not dependent on any other code or service outside solana-py library, which is why I put it as a unit tests. 

I can still move to integration tests if you prefer this, but I wanted to point out these tests have mocking for request before doing so.

**Review by michaelhly — APPROVED** [review]

**Review by kevinheavey — APPROVED** [review]

**Pull request merged**